tests/features/bootstrap/FeatureContext.php

```diff
@@ -77,41 +80,44 @@ public function iAmLoggedInAsAnAdmin()
     {   $this->visitPath("/admin.php");
         $this->iLoginWithUsernameAndPassword("admin","mypass");
     }
-    /**
-     * @When I goto Add a New Article
-     */
-    public function iGotoAddaNewArticle()
-    {
-        $this->getSession()->getPage()->find('xpath','//a[@href="admin.php?action=newArticle"]')->click();
-    }
-
-    /**
-     * @Given I fill in the following details
-     */
-    public function iFillTheFollowingDetails(TableNode $table)
-    {
-        $page=$this->getSession()->getPage();
-        $tableArray = $table->getColumnsHash();
-        $page->fillField("title",$tableArray[0]["title"]);
-        $page->fillField("summary", $tableArray[0]["summary"]);
-        $page->fillField("content", $tableArray[0]["content"]);
-        $page->fillField("publicationDate", $tableArray[0]["date"]);
-    }
-    /**
-     * @Given I save the changes
-     */
-    public function iSaveTheChanges()
-    {
-     $page=$this->getSession()->getPage();
-     $page->find('xpath', '//div/input[@name="saveChanges"]')->click();
-     }
-    /**
-     * @Given I am on the New Article page
-     */
-    public function iAmOnTheNewArticlePage()
-    {
-        $this->visitPath("/admin.php?action=newArticle");
-    }
+	/**
+	 * @When I goto Add a New Article
+	 * 
+	 * @return void
+	 */
+	public function iGotoAddaNewArticle() {
+		$this->getSession()->getPage()->find('xpath', '//a[@href="admin.php?action=newArticle"]')->click();
+	}
+	
+	/**
+	 * @Given I fill in the following details
+	 * 
+	 * @param TableNode $table
+	 * @return void
+	 */
+	public function iFillTheFollowingDetails(TableNode $table) {
+		$tableArray = $table->getColumnsHash();
+		$this->addarticle->addTitle($tableArray);
```

Comment:
We should not need to pass the whole ``$tableArray`` to ``addTtitle``
``addTitle`` only needs to know the text to put in the title field. It should not need to know the format of ``$tableArray`` and how to find the title text inside that array.

Similar for ``addSummary`` ``addContent`` ``addDate``